### ISSUE TITLE
Video thumbnail previews with chafa

### ISSUE THREAD

**Issue by Zerogaku** [gokcehan/lf#1662]
Hi I was wondering if it's possible to have video thumbnail previews using chafa preview in lf?

**Comment by prettykool** [comment]
If you have `ffmpegthumbnailer` installed, you can do it without a sweat. I've been using this setup for a while now:
```sh
# In your script for creating previews
case "$file" in
*.avi|*.gif|*.mp4|*.mkv|*.webm)
    ffmpegthumbnailer -i "$file" -s 0 -q 5 -o "/tmp/thumbnail.png"
    preview "$thumbnail" "$@"
    ;;
esac
```
![image](https://github.com/gokcehan/lf/assets/79608594/ac70479d-fe02-4181-b765-dc60b87603ca)

**Comment by DusanLesan** [comment]
Is that app cycling through several frames off the video or staying at a single one? Having single frame displayed would be easy with ffmpeg

**Comment by prettykool** [comment]
> @prettykool Is the app cycling through several frames of the video or staying at a single frame? Displaying a single frame would be easy with ffmpeg.

It just displays a single frame from the video. What solution do you have in mind?

**Comment by DusanLesan** [comment]
> It just displays a single frame from the video. What solution do you have in mind?

I don't have anything specific in mind. I was just hoping for something more from app specialised for the task.

**Comment by Zerogaku** [comment]
@prettykool Thank for the help! I needed to add some missing code for it to work properly though:
```
preview() {
    chafa "$@" -f sixel -s "$(($2-2))x$3" | sed 's/#/\n#/g'
}

thumbnail="/tmp/thumbnail.png"

file="$1"; shift

case "$file" in
*.avi|*.gif|*.mp4|*.mkv|*.webm)
    ffmpegthumbnailer -i "$file" -s 0 -q 5 -o "/tmp/thumbnail.png"
    preview "$thumbnail" "$@"
    ;;
esac
```
I also found [this](https://github.com/horriblename/lfimg-sixel/blob/master/preview) slightly more complex version of the preview script which handles caching/storing the generated video thumbnails.

@DusanLesan https://github.com/gokcehan/lf/issues/1509#issuecomment-1826736409 might answer that question, the preview script is supposed to output something very quickly and then terminate. I think it is possible to have video thumbnails or gifs cycle, but said script would need to be more complex, like for example generate 5 images from certain points the the video and store them in a cache directory, and for loop through those images, and when a new video is chosen, terminate the previous loop and start another, though I could be wrong.
